src/app/shared/services/remote-asset/remote-asset.service.spec.ts

```diff
@@ -1,21 +1,56 @@
 import { TestBed } from "@angular/core/testing";
 
 import { RemoteAssetService } from "./remote-asset.service";
-import { AppDataService } from "../data/app-data.service";
+import { HttpClientTestingModule } from "@angular/common/http/testing";
+import { IAssetContents } from "src/app/data";
+import { FlowTypes } from "../../model";
 
+const MOCK_ASSETS_CONTENTS_LIST: IAssetContents = {
+  "audio/test_audio.mp3": {
+    md5Checksum: "5ddddf934d2187d084c75b7e27797fae",
+    size_kb: 43.4,
+  },
+};
+
+const MOCK_CORE_ASSET_PACK: FlowTypes.AssetPack = {
+  flow_type: "asset_pack",
+  flow_name: "core_assets",
+  rows: [
+    {
+      id: "audio/test_audio.mp3",
+      md5Checksum: "5ddddf934d2187d084c75b7e27797fae",
+      size_kb: 43.4,
+    },
+  ],
+  rowsHashmap: {
+    "audio/test_audio.mp3": {
+      id: "audio/test_audio.mp3",
+      md5Checksum: "5ddddf934d2187d084c75b7e27797fae",
+      size_kb: 43.4,
+    },
+  },
+};
+
+/**
+ * Call standalone tests via:
+ * yarn ng test --include src/app/shared/services/remote-asset/remote-asset.service.spec.ts
+ */
 describe("RemoteAssetsService", () => {
   let service: RemoteAssetService;
 
   beforeEach(() => {
-    TestBed.configureTestingModule({});
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+    });
     service = TestBed.inject(RemoteAssetService);
   });
 
   it("should be created", () => {
     expect(service).toBeTruthy();
   });
 
-  it("generates core asset pack from asset contents", () => {
-    // TODO
+  it("generates a core asset pack from asset contents", () => {
+    const coreAssetPack = service["generateCoreAssetPack"](MOCK_ASSETS_CONTENTS_LIST);
```

Comment:
nit(non-blocking): any particular reason to access the method in this way instead of just `service.generateCoreAssetPack`? Doesn't make any functional difference, just curious

Reply:
This was a way I found of accessing a private method in the test. I'm not quite sure if it's a common or desirable pattern but it seems to work for cases like this.
Have you got any thoughts about testing private methods? I know that it's often suggested that only public methods should have unit tests, but in a case like this it seems like a good idea to have a test for this method, and the only reason to make it public would be to make it available for testing. I suppose it could live elsewhere as a util and have an associated unit test, but it is specifically a remote assets/asset pack util so it seems right that it should sit in the service file to me.

Reply:
Oh, interesting! I think I remember trying similar before, cool that it just works like that.

As a general rule of thumb I'd say most of the time can assume private methods will be called somewhere from public methods, and so can usually be tested indirectly. In this case the private method is called during a separate private init so might be harder to track except using the methods mocked from the `templateAssetService` which is reads from and updates as part of the init workflow. 

Utils are another way to go, although this method doesn't really do much in the lines of manipulation (converts hashmap to array, which possibly we could add general util for given we have `arrayToHashmap`, and populates as a flow type). So I think for this case I'd lean in to making the `hashmapToArray` a util with spec tests, moving the side-effect code `addRuntimeFlowToContents` to the init flow instead of within the function call, and skipping the need to test that it essentially just populates the generated list within the flow structure (as typescript would give error if that structure changed). 

But more generally I don't have any issue with what you've done - tests are useful to have, I agree best to not change core code just to accomodate tests (unless no reasonable way around), and particularly if code gets longer/more complex adding intermediate tests for private methods still makes sense in theory to me.